core/netparams/defaults.go

```diff
@@ -217,7 +217,7 @@ func defaultNetParams() map[string]value {
 		NetworkCheckpointTimeElapsedBetweenCheckpoints: NewDuration(DurationGT(0 * time.Second)).Mutable(true).MustUpdate("1m"),
 		// take a snapshot every 1000 blocks, ~20 minutes
 		// if we assume a block time of anything between 1 to 2 seconds
-		SnapshotIntervalLength: NewInt(gteI0).Mutable(true).MustUpdate("1000"),
+		SnapshotIntervalLength: NewUint(gteU1).Mutable(true).MustUpdate("1000"),
```

Comment:
Though in practice we probably never want to do anything of the sort, but by restricting this value to be >= 1, we effectively make it impossible to snapshot every block should we ever want to do that